### DISCUSSION TITLE
webauthn4j for client side

### DISCUSSION THREAD

**Discussion by zarashima** [webauthn4j/webauthn4j#1061] [Q&A] [answered]
Hi,

I'd like to conduct a performance test for FIDO2 APIs using Jmeter, but not sure what kind of available Webauthn library to use.
Any suggestions or advice?

Thanks

**Comment by ynojima** [comment, marked as answer]
Interesting question.
For E2E test, you can use "WebAuthn WebDriver Extension", which is supported by Chrome and selenium: https://github.com/SeleniumHQ/selenium/issues/10541
For performance test, Chrome + selenium may be too heavy, and webauthn authenticator emulator plugin for load generator like jmeter might be needed.
But I assume no webauthn plugin exists.
Since webauthn4j-test module provides webauthn authenticator emulator classes to be used for webauthn4j integration testing, it might be good fit for developing jmeter plugin (As I'm not familiar with jmeter, I'm not sure if it is possible with jmeter plugin architechture).

This is the test code that uses the webauthn authenticator emulator of `webauthn4j-test`:
https://github.com/webauthn4j/webauthn4j/blob/master/webauthn4j-core/src/test/java/integration/scenario/webauthn/AttestationVariationTest.java#L69-L70
